### ISSUE TITLE
TerminalRefreshCustomization Not Working

### ISSUE THREAD

**Issue by kmccracko** [darmuh/TerminalStuff#35]
After changing terminal customizations (colors, home screen text, etc), I expected the "paintme" command to reapply the config values in-game, making customization easier than restarting the game each time. However, I just get the message that it's happening (`Refreshing TerminalCustomization from config.`), with no changes from what I can tell.

I do have "TerminalRefreshCustomization" enabled, and I've tried refreshing through both the terminal commands and the LethalConfig button.

With extensive and developer logs enabled, here's the output when I enter "paintme":

```
[Info   :darmuhsTerminalStuff] disabling ANY cams views
[Info   :darmuhsTerminalStuff] Networked nodes enabled, sending result to server.
[Info   :darmuhsTerminalStuff] Valid node detected, nNS true
[Info   :darmuhsTerminalStuff] RPC called from host, sending to client RPC
[Info   :darmuhsTerminalStuff] Node detected coming from host, resetting nNS and ending RPC
[Info   :darmuhsTerminalStuff] LoadNewNode patch, nNS: False
```

Interestingly, no logs are output when clicking the Refresh button in LethalConfig.

This could also just be stemming from a misunderstanding of what the command is for. Would appreciate it if you could shed some light!

**Comment by darmuh** [comment]
Hello @kmccracko,

Your understanding of the command is correct. I'm not sure why it is not working for you. The reason you are not seeing anything in logs is because there is not much logging in the [method itself](https://github.com/darmuh/TerminalStuff/blob/f7531381d22657e48256185a3f40d3aa316ed2a5/DarmuhsTerminalCommands/PluginCore/TerminalCustomizer.cs#L96) other than when issues are identified during customization.

I can also look into this further if you provide a profile code for me to test with on my end.

Also, I'd suggest you make sure the TerminalCustomization setting in the section "Terminal Customization" is enabled. As all methods related to terminal customizations will halt when this is set to false.

**Comment by kmccracko** [comment]
Thanks for the quick reply!

I've checked and TerminalCustomization is enabled as well. On more thorough testing, what I'm finding is that the color values (like the physical terminal and the terminal text) do correctly respond to the refresh. Yay! However, the text content itself (like the homelines and the art) don't update, and that was actually my use case for the command, hence my not thinking it was working.

So to clear things up again, is the terminal refresh supposed to refresh everything in the "Terminal customization" menu? If so, it isn't working 100% as expected. If it's only for the color values, then it's working great!

If this isn't expected, here's a test profile for further investigation: `01927cbe-87a0-8810-ce58-37dc72e6b08e`

To reproduce,
1. Open up lethal config in the menu and select your mod
2. Adjust text like "Welcome" to "Hello"
3. Adjust terminal color to #FF0000
4. Apply changes
5. "Refresh" button, then check terminal
-- terminal color is red
-- text still says "Welcome"

---

_**Edited below to also mention the MirrorDecor mod_
By the way, I'm encountering issues (not in the profile I shared above) with the "mirror" feature of the terminal showing only character arms, or sometimes no character at all. I think the issue is the ModelReplacementAPI mod but am still testing a bit. Are you aware of any conflicts like that?

I think I have it pretty conclusively now:
1. Having the [MirrorDecor](https://thunderstore.io/c/lethal-company/p/quackandcheese/MirrorDecor/) mod causes the ghosting/disappearing.
2. Having the [ModelReplacementAPI](https://thunderstore.io/c/lethal-company/p/BunyaPineTree/ModelReplacementAPI/), mod causes only the models's arms to be visible.
3. Having neither mod shows the character/suit properly as expected.

Apologies for talking about things not relevant to this issue, but I wasn't certain I should put in a new issue for something that's only an issue by interaction with other mods. If you want, I can open a new issue for this, but otherwise just something to know in case other questions come up.

**Comment by darmuh** [comment]
Yes currently the refresh for customizations only applies to colors (even though they are in the same section). I will see if I can update this in the future.

In regards to the camera issue, I'm aware of the issue and have some work in development for this. Unfortunately a lot of mods like to touch the player object layers which can cause these issues with the camera culling mask. In the meantime, if you enable "CamsUseDetectedMods" alongside OpenBodyCams I believe that version of the camera will not have these issues. (Not sure about MirrorDecor specifically, but I am certain for modelreplacementapi)

**Comment by kmccracko** [comment]
Okay great, thank you for clearing up the refresh command for me.

I do have "CamsUseDetectedMods" enabled, but I guess it doesn't work for the mirror without OpenBodyCams installed. This isn't a huge issue to me, so I think I'll forgo the additional mod. Thanks again for the your responsiveness. Love what you've done!

**Comment by darmuh** [comment]
Ah yes I'm only able to create the mirror camera with one external mod (OpenBodyCams). I dont think I'd be able to do the same with Helmet Cameras or Solo's BodyCams which are the other two external mods I know of.

Speaking of which, I have pushed an update today with the fixes for my homebrew cams. I also added the ability to update the home page text when refreshing terminal customizations. Thanks for using my mod!
